src/server/scripts/Northrend/Gundrak/boss_moorabi.cpp

```diff
@@ -125,41 +138,49 @@ class boss_moorabi : public CreatureScript
                 }
             }
 
-            void ExecuteEvent(uint32 eventId) override
+            void UpdateAI(uint32 diff) override
             {
-                switch (eventId)
+                if (!me->IsInCombat())
+                    scheduler.Update(diff);
```

Comment:
Nope, you can control it by event phase xD
e.g.
If you need that 1 event only happens before combat:

On reset:

events.SetPhase(PHASE_INTRO);

OnUpdateAI:

if (!UpdateVictim() && !events.InPhase(PHASE_INTRO));

OnEnterCombat:

events.SetPhase(PHASE_COMBAT);